Mathlib/Algebra/Category/GroupCat/Injective.lean

```diff
@@ -44,379 +44,148 @@ set_option linter.uppercaseLean3 false
 namespace AddCommGroupCat
 
 theorem injective_of_injective_as_module [Injective (⟨A⟩ : ModuleCat ℤ)] :
-    CategoryTheory.Injective (⟨A,inferInstance⟩ : AddCommGroupCat) :=
-  { factors := fun {X} {Y} g f m => by
-      let G : (⟨X⟩ : ModuleCat ℤ) ⟶ ⟨A⟩ :=
-        { g with
-          map_smul' := by
-            intros
-            dsimp
-            rw [map_zsmul] }
-      let F : (⟨X⟩ : ModuleCat ℤ) ⟶ ⟨Y⟩ :=
-        { f with
-          map_smul' := by
-            intros
-            dsimp
-            rw [map_zsmul] }
-      have : Mono F := by
-        refine' ⟨fun {Z} α β eq1 => _⟩
-        -- Porting note: trouble getting to ℤ-module from ModuleCat ℤ
-        -- AddCommGroup.intModule not defeq to .isModule
-        let α' : AddCommGroupCat.of Z ⟶ X := @LinearMap.toAddMonoidHom _ _ _ _ _ _ _ _ (_) _ _ α
-        let β' : AddCommGroupCat.of Z ⟶ X := @LinearMap.toAddMonoidHom _ _ _ _ _ _ _ _ (_) _ _ β
-        have eq2 : α' ≫ f = β' ≫ f := by
-          ext x
-          simp only [CategoryTheory.comp_apply, LinearMap.toAddMonoidHom_coe]
-          simpa only [ModuleCat.coe_comp, LinearMap.coe_mk, Function.comp_apply] using
-            FunLike.congr_fun eq1 x
-        rw [cancel_mono] at eq2
-        have : ⇑α' = ⇑β' := congrArg _ eq2
-        ext x
-        apply congrFun this _
-      refine' ⟨(Injective.factorThru G F).toAddMonoidHom, _⟩
-      ext x
-      convert FunLike.congr_fun (Injective.comp_factorThru G F) x}
+    CategoryTheory.Injective (⟨A,inferInstance⟩ : AddCommGroupCat) where
```

Comment:
In fact, we know that `ModuleCat \Z` and `AddCommGroupCat` are equivalent categories. So this lemma should really follow from a generic lemma that says that equivalences preserve injectivity of objects. Would you mind factoring out that lemma? You could reuse it in the lemma below this one.

Reply:
Great idea, [golfed](https://github.com/leanprover-community/mathlib4/pull/7204/commits/d95b6e47e07761256a5e884289e2495e0cf830c4).

> a generic lemma that says that equivalences preserve injectivity of objects

There are `Adjunction` lemmas directly applicable so I think there's no point adding a `Equivalence` version.

Reply:
I added the Equivalence anyway: https://github.com/leanprover-community/mathlib4/pull/7204/commits/a2bebbfb7e74c9f40d1fc012b2b8cc5387dc0cfe